peer.go

```diff
@@ -666,8 +666,89 @@ out:
 			isChanUpdate = true
 			targetChan = msg.ChanID
 
-		case *lnwire.ChannelUpdate,
```

Comment:
Can you explain why this was changed and ProcessRemoteAnnouncement cannot be used?

Reply:
In `AnnounceSignatures`, two nodes exchange `ChannelEdgePolicy` and update their peer's directed edges in their own channel graph. Now, per issue #306, it is possible that they don't want their channel announced to the greater network. This means not sending an `AnnounceSignatures` message. But there must be some way of exchanging `ChannelUpdate` messages, so I altered `peer.go` to include functionality to determine when it is acceptable to pass it to the gossiper, and when it is acceptable to NOT (to preserve privacy).

Reply:
I was also told by @Roasbeef to not use flags in the `gossiper.go` to gate such behavior... I know that `peer.go` isn't pretty, but I wasn't sure what else to do.

Reply:
I think you are right that I can use `ProcessRemoteAnnouncement` ... but I'm less sure about using it for `AddEdge` & `UpdateEdge`

Reply:
I suppose I can do this in `ProcessRemoteAnnouncement`:

1) If we receive a `ChannelAnnouncement`, `isRemote` is false and `ChannelEdgeInfo` is not returned from `GetChannelByID`, we don't announce & we call `Router.AddEdge`.
2) If we receive a `ChannelUpdate`, `isRemote` is false and no `ChannelEdgePolicy` is returned from `GetChannelByID`, we don't announce & we call `Router.UpdateEdge`. We can just not include an authentication proof here and it won't be announced.

Let me think about `AnnounceSignatures` and get back to you.

Reply:
We can also just include the `AuthProof` in step 2) above.
This would mean that the `ChannelUpdate` would be sent to the remote peer and using similar logic to the logic you commented about would mean the `AnnounceSignatures` branch wouldn't need to be called. Let me know what you think.

Reply:
My above comment is a bit wrong, we shouldn't include the `AuthProof` in step 2).  Instead if the code follows the branch that doesn't announce `ChannelUpdate`, this code branch should also send the `ChannelUpdate` message directly to the peer so there is no announcement happening.

Reply:
Alright I reverted the changes on `peer.go` and the relevant code is now in `gossiper.go`.  Let me know what you think.  It works as before, but this is better I believe so thanks for the comments!